src/main.rs

```diff
@@ -165,6 +260,196 @@ fn create_pipeline(app: &App) -> Result<(gst::Pipeline, gtk::Widget), Box<dyn er
     Ok((pipeline, widget))
 }
 
+fn save_settings(
+    folder_path: &gtk::Entry,
+    options: &gtk::ComboBoxText,
+    timer_entry: &gtk::SpinButton,
+) {
+    let settings = SnapshotSettings {
+        folder: PathBuf::from(folder_path.get_text().unwrap_or_else(|| ".".to_owned())),
+        format: OutputFormat::from(options.get_active_text()),
+        timer_length: timer_entry.get_value_as_int() as _,
+    };
+    let s = get_settings_file_path();
+    if let Err(e) = serde_any::to_file(&s, &settings) {
+        eprintln!("Error when trying to save file: {:?}", e);
+    } else {
+        println!("Saved settings {:?} in '{}'", settings, s.display());
+    }
+}
+
+fn build_screenshot_settings_window(parent: &gtk::Window) {
+    let s = get_settings_file_path();
+
+    if !s.exists() {
+        if let Some(parent) = s.parent() {
+            if !parent.exists() {
+                if let Err(e) = create_dir_all(parent) {
+                    eprintln!("Error when trying to build settings folder '{}': {:?}",
+                              parent.display(),
+                              e);
+                }
+            }
+        }
+    }
+
+    let settings = if s.exists() && s.is_file() {
+        match serde_any::from_file::<SnapshotSettings, _>(&s) {
+            Ok(s) => s,
+            Err(e) => {
+                eprintln!("Error when opening '{}': {:?}", s.display(), e);
+                SnapshotSettings::default()
+            }
+        }
+    } else {
+        SnapshotSettings::default()
+    };
+
+    //
+    // BUILDING UI
+    //
+    let dialog = gtk::Dialog::new_with_buttons(Some("Screenshot settings"),
+                                               Some(parent),
+                                               gtk::DialogFlags::MODAL,
+                                               &[]);
+
+    let grid = gtk::Grid::new();
+    grid.set_column_spacing(4);
+    grid.set_row_spacing(4);
+
+    //
+    // OUTPUT FOLDER
+    //
+    let folder_label = gtk::Label::new("Output folder");
+    let folder_hbox = gtk::Box::new(gtk::Orientation::Horizontal, 1);
+    let folder_chooser_but = gtk::Button::new_from_icon_name("folder", 1);
```

Comment:
Or a `gtk::FileChooserButton`?

Reply:
That also comes directly with the dialog built-in :)

Reply:
Urg... That makes things so much easier!